src/Swarm/TUI/Model.hs

```diff
@@ -255,9 +262,19 @@ uiTPF :: Lens' UIState Double
 -- | Computed frames per milli seconds
 uiFPS :: Lens' UIState Double
 
--- | The base-2 logarithm of the current game speed in ticks per
---   second.
+-- | The base-2 logarithm of the current game speed in ticks/second.
+--   Note that we cap this value to the range of +/- log2 INTMAX.
 lgTicksPerSecond :: Lens' UIState Int
+lgTicksPerSecond = lens _lgTicksPerSecond safeSetLgTicks
+ where
+  maxLog = floor (logBase 2 (fromIntegral (maxBound :: Int)) :: Float)
```

Comment:
What do you think about using `finiteBitSize` from `Data.Bits` instead of `logBase 2` etc? Seems cleaner to me.